File: samplesJS/Http/assets/scripts/HttpSample.js
var font = getFont("font.fnt");

var originIP = "";
var userAgent = "";
var downloadedTexture = null;
var postResult = "";

function update(dt) {
    if (Input.isJustDown(Key._1)) {
        // Stalls the main loop
        originIP = Http.getString("http://httpbin.org/ip");
    }
    if (Input.isJustDown(Key._2)) {
        // Doesn't stall the main loop
        Http.getStringAsync("http://httpbin.org/user-agent", {}, function (result) {
            userAgent = result;
        });
    }
    if (Input.isJustDown(Key._3)) {
        Http.getTextureAsync("https://upload.wikimedia.org/wikipedia/commons/thumb/4/47/PNG_transparency_demonstration_1.png/560px-PNG_transparency_demonstration_1.png", {}, function (texture) {
            downloadedTexture = texture;
        });
    }
    if (Input.isJustDown(Key._4)) {
        postResult = Http.post("http://httpbin.org/post", {key:"Hello World!"});
    }
}

function render() {
    // Clear
    Renderer.clear(Color.fromHexRGB(0x1d232d));

    SpriteBatch.begin();

    SpriteBatch.drawText(font, "Press ^9911^999 to GET Origin IP.", new Vector2(10, 10));
    SpriteBatch.drawText(font, originIP, new Vector2(10, 30), Vector2.TOP_LEFT, new Color(.7, .7, .7, 1));

    SpriteBatch.drawText(font, "Press ^9912^999 to GET User Agent Asynchronously.", new Vector2(10, 110));
    SpriteBatch.drawText(font, userAgent, new Vector2(10, 130), Vector2.TOP_LEFT, new Color(.7, .7, .7, 1));

    SpriteBatch.drawText(font, "Press ^9913^999 to download a picture.", new Vector2(10, 210));
    if (downloadedTexture)
        SpriteBatch.drawRect(downloadedTexture, new Rect(10, 230, downloadedTexture.getSize()));

    SpriteBatch.drawText(font, "Press ^9914^999 to POST.", new Vector2(400, 10));
    SpriteBatch.drawText(font, postResult, new Vector2(400, 30), Vector2.TOP_LEFT, new Color(.7, .7, .7, 1));

    SpriteBatch.end();
}

File: samples/Http/src/HttpSample.cpp
// Oak Nut include
#include <onut/Font.h>
#include <onut/Http.h>
#include <onut/Input.h>
#include <onut/Renderer.h>
#include <onut/Settings.h>
#include <onut/SpriteBatch.h>
#include <onut/Texture.h>

std::string originIP;
std::string userAgent;
OTextureRef pDownloadedTexture;
std::string postResult;

void initSettings()
{
    oSettings->setGameName("HTTP Sample");
}

void init()
{
}

void update()
{
    if (OInputJustPressed(OKey1))
    {
        // Stalls the main loop
        originIP = OHttpGetString("http://httpbin.org/ip");
    }
    if (OInputJustPressed(OKey2))
    {
        // Doesn't stall the main loop
        OHttpGetStringAsync("http://httpbin.org/user-agent", [](std::string result)
        {
            userAgent = result;
        });
    }
    if (OInputJustPressed(OKey3))
    {
        OHTTPGetTextureAsync("https://upload.wikimedia.org/wikipedia/commons/thumb/4/47/PNG_transparency_demonstration_1.png/560px-PNG_transparency_demonstration_1.png", [](OTextureRef pTexture)
        {
            pDownloadedTexture = pTexture;
        });
    }
    if (OInputJustPressed(OKey4))
    {
        const char* szData = "Hello World!";
        OHttp::Body body(szData, szData + strlen(szData) + 1);
        postResult = OHttpPost("http://httpbin.org/post", body);
    }
}

void render()
{
    // Clear
    oRenderer->clear(OColorHex(1d232d));

    // Draw Info
    auto pFont = OGetFont("font.fnt");

    oSpriteBatch->begin();

    pFont->draw("Press ^9911^999 to GET Origin IP.", {10, 10});
    pFont->draw(originIP, {10, 30}, OTopLeft, Color(.7f, .7f, .7f, 1));

    pFont->draw("Press ^9912^999 to GET User Agent Asynchronously.", {10, 110});
    pFont->draw(userAgent, {10, 130}, OTopLeft, Color(.7f, .7f, .7f, 1));

    pFont->draw("Press ^9913^999 to download a picture.", {10, 210});
    if (pDownloadedTexture)
        oSpriteBatch->drawRect(pDownloadedTexture, Rect(10, 230, pDownloadedTexture->getSizef()));

    pFont->draw("Press ^9914^999 to POST.", {400, 10});
    pFont->draw(postResult, {400, 30}, OTopLeft, Color(.7f, .7f, .7f, 1));

    oSpriteBatch->end();
}

void postRender()
{
}

void renderUI()
{
}
